### DISCUSSION TITLE
Feature Request: Interruptible Cipher Operations with Progress Callbacks

### DISCUSSION THREAD

**Discussion by landabaso** [paulmillr/noble-ciphers#23] [Ideas]
I'd like to propose an enhancement for `noble-ciphers` to support interruptible cipher operations with progress callbacks, particularly beneficial for processing very long data in environments like React Native and web.

The library is excellent; however, I'm encountering issues with large datasets causing the main execution thread to stall for a few seconds, resulting in a slightly unresponsive appearance for the app.

This feature would allow passing an optional callback function to cipher methods, which would be invoked approximately every 100ms (user defined). This callback could provide progress updates (if possible) and offer a mechanism to interrupt the ciphering process if needed, enhancing usability for large datasets and improving application responsiveness.

One possible implementation approach could involve maintaining a timestamp in the main event loop to track the last time the progress callback was invoked.

**Comment by paulmillr** [comment]
Why don't you use an async webworker?

**Reply by landabaso** [reply]
Yup, while async web workers could be a viable option for web development, my current project involves React Native app development. Unfortunately, React Native doesn't have built-in support for web workers. Additionally, I've explored other thread-based libraries in React Native, but they often come with their own set of limitations, such as only working for simple examples like the "fibonacci" sequence and not allowing the calling of external functions, or being unmaintained.

Apart from responsiveness, having a callback with percentage progress and cancellable functionality may be interesting to others as well.

If you have any other ideas or suggestions for achieving similar functionality in a React Native environment, I'm definitely open to exploring them!
